renku_notebooks/api/schemas/cloud_storage.py

```diff
@@ -29,80 +30,153 @@ def validate_storage(self, data, **kwargs):
-            credential=configuration["secret_access_key"],
-            mount_folder=str(work_dir / target_path),
-            source_folder=source_path,
-            read_only=readonly,
-        )
-    elif configuration.get("type") == "s3" and config.cloud_storage.s3.enabled:
-        cloud_storage = S3Request(
-            endpoint=configuration.get("endpoint"),
-            region=configuration.get("region"),
-            bucket=bucket,
-            access_key=configuration.get("access_key_id"),
-            secret_key=configuration.get("secret_access_key"),
-            mount_folder=work_dir / target_path,
-            source_folder=source_path,
-            read_only=data.get("readonly", True),
-        )
-    else:
-        raise ValidationError(
-            "Cannot accept the provided cloud storage parameters because "
-            "the requested storage type has not been properly setup or enabled."
+class RCloneStorage:
+    def __init__(
+        self,
+        source_path: str,
+        configuration: Dict[str, Any],
+        readonly: bool,
+        mount_folder: str,
+        name: Optional[str],
+    ) -> None:
+        config.storage_validator.validate_storage_configuration(configuration, source_path)
+        self.configuration = configuration
+        self.source_path = source_path
+        self.mount_folder = mount_folder
+        self.readonly = readonly
+        self.name = name
+
+    @classmethod
+    def storage_from_schema(cls, data: Dict[str, Any], user: User, project_id: int, work_dir: Path):
+        name = None
+        if data.get("storage_id"):
+            # Load from storage service
+            if user.access_token is None:
+                raise ValidationError("Storage mounting is only supported for logged-in users.")
+            if project_id < 1:
+                raise ValidationError("Could not get gitlab project id")
+            (
+                configuration,
+                source_path,
+                target_path,
+                readonly,
+                name,
+            ) = config.storage_validator.get_storage_by_id(user, project_id, data["storage_id"])
+            configuration = {**configuration, **(configuration or {})}
+            readonly = readonly
+        else:
+            source_path = data["source_path"]
+            target_path = data["target_path"]
+            configuration = data["configuration"]
+            readonly = data.get("readonly", True)
+        mount_folder = str(work_dir / target_path)
+
+        return cls(source_path, configuration, readonly, mount_folder, name)
+
+    def get_manifest_patch(
+        self, base_name: str, namespace: str, labels={}, annotations={}
+    ) -> List[Dict[str, Any]]:
+        patches = []
+        patches.append(
+            {
+                "type": "application/json-patch+json",
+                "patch": [
+                    {
+                        "op": "add",
+                        "path": f"/{base_name}-pv",
```

Comment:
Reminder to take this out when the CSI driver is ready. Also we are not merging this like this until the driver is ready, correct?

Reply:
we're not releasing it without that. not sure about merging